service/app/controllers/Wishlist.scala

```diff
@@ -0,0 +1,109 @@
+package controllers
+
+import com.google.inject.Inject
+import commons.{CollectionLinks, Error, ErrorResponse}
+import javax.inject.Singleton
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, AnyContent, Controller, Result}
+import repositories.{ProductRepository, WishlistItemsRepository}
+import wishlist.{WishlistItemCollectionResponse, WishlistItemRequest, WishlistItemResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class Wishlist @Inject()(wishlists: WishlistItemsRepository,
+                         products: ProductRepository,
+                         secure: SecuredAuthenticator)(implicit val ec: ExecutionContext)
+    extends Controller {
+
+  import hateoas.JsonApi._
+  import secure.Roles._
+
+  def retrieve(userId: Long): Action[AnyContent] = secure.AuthWith(Seq(Customer)).async {
+    implicit request =>
+      if (request.user.isDefined && request.user.get.id.get != userId) {
```

Comment:
Should it be `request.user.isEmpty || request.user.get.id.get != userId`?
Should it fail id `request.user` is undefined?